kubernetes/e2e_test/test_utils.py

```diff
@@ -112,5 +112,85 @@ def test_api_service(self):
         svc = k8s_api.read_api_service(
             name="v1alpha1.wardle.k8s.io")
         self.assertIsNotNone(svc)
+        svc_conflict = utils.create_from_yaml(k8s_client,
+            "kubernetes/e2e_test/test_yaml/api-service.yaml")
+        self.assertEqual([], svc_conflict)
         resp = k8s_api.delete_api_service(
-            name="v1alpha1.wardle.k8s.io", body={})
\ No newline at end of file
+            name="v1alpha1.wardle.k8s.io", body={})
+
+    def test_list(self):
+        k8s_client = client.api_client.ApiClient(configuration=self.config)
+        k8s_api = utils.create_from_yaml(k8s_client, 
+            "kubernetes/e2e_test/test_yaml/list.yaml")
+        svc_api = k8s_api[0]
+        self.assertEqual("v1", svc_api.get_api_resources().group_version)
+        svc = svc_api.read_namespaced_service(name="list-service-test",
+            namespace="default")
+        self.assertIsNotNone(svc)
+        ext_api = k8s_api[1]
+        self.assertEqual("extensions/v1beta1", 
+            ext_api.get_api_resources().group_version)
+        dep = ext_api.read_namespaced_deployment(name="list-deployment-test",
+            namespace="default")
+        self.assertIsNotNone(dep)
+        ext_api.delete_namespaced_deployment(name="list-deployment-test",
+            namespace="default", body={})
+        svc_api.delete_namespaced_service(name="list-service-test",
+            namespace="default", body={})
+    
+    def test_multi_resource(self):
+        k8s_client = client.api_client.ApiClient(configuration=self.config)
+        k8s_api = utils.create_from_yaml(k8s_client, 
+            "kubernetes/e2e_test/test_yaml/multi-resource-yaml.yaml")
+        svc_api = k8s_api[0]
+        self.assertEqual("v1", svc_api.get_api_resources().group_version)
+        svc = svc_api.read_namespaced_service(name="mock",
+            namespace="default")
+        self.assertIsNotNone(svc)
+        ctr_api = k8s_api[1]
+        self.assertEqual("v1", ctr_api.get_api_resources().group_version)
+        ctr = ctr_api.read_namespaced_replication_controller(
+            name="mock", namespace="default")
+        self.assertIsNotNone(ctr)
+        ctr_api.delete_namespaced_replication_controller(name="mock",
+            namespace="default", body={})
+        svc_api.delete_namespaced_service(name="mock",
+            namespace="default", body={})
+
+    def test_multi_resource_with_conflict(self):
```

Comment:
need to document the tests. it's hard to follow what's expected from reading the test and file names